server/fileserver.go

```
@@ -72,3 +119,43 @@ func getUploadHandler(fsPath string) http.Handler {
 		logger.Info("Finished uploading")
 	})
 }
+
+func getDownloadHandler(fsPath string) http.Handler {
+	return http.HandlerFunc(func (res http.ResponseWriter, req *http.Request) {
+		var (
+			status int
+			err  error
+		)
+		defer func() {
+			if err != nil {
+				writeResponse(res, req, status, &Response{err.Error()})
+				return
+			}
+		}()
+		path := filepath.Join(fsPath,req.URL.String())
+		//first check if its folder or file.
+		info, err := os.Stat(path)
+		if os.IsNotExist(err) {
+			logger.WithError(err).Error("file/folder does not exist")
+			status = http.StatusNotFound
+			return
+		}
+		if info.IsDir() {
+			path = req.URL.String()
+			tarFileName := filepath.Base(path)
+			fullPathToTar := filepath.Join(filepath.Join(fsPath, path,tarFileName)) + ".tar.gz"
+			tarFile, err := os.Create(fullPathToTar)
```

do not ignore the error returned from os.Create